src/main/java/net/neoforged/neoforge/common/extensions/IAttributeExtension.java

```diff
@@ -0,0 +1,217 @@
+        return Component.translatable(key, FORMAT.format(isNullOrAddition(op) ? value : value * 100));
+    }
+
+    /**
+     * Converts an attribute modifier into its tooltip representation.
+     * <p>
+     * This method does not handle formatting of "base" modifiers, such as Attack Damage or Attack Speed.
+     * <p>
+     * The returned component may append additional debug information based on the tooltip flag.
+     *
+     * @param modif The attribute modifier being converted to a component.
+     * @param flag  The tooltip flag.
+     * @return The component representation of the passed attribute modifier, with debug info appended if enabled.
+     */
+    default MutableComponent toComponent(AttributeModifier modif, TooltipFlag flag) {
+        Attribute attr = self();
+        double value = modif.amount();
+
+        MutableComponent comp;
+
+        if (value > 0.0D) {
+            comp = Component.translatable("neoforge.modifier.plus", this.toValueComponent(modif.operation(), value, flag), Component.translatable(attr.getDescriptionId())).withStyle(attr.getStyle(true));
+        } else {
+            value *= -1.0D;
+            comp = Component.translatable("neoforge.modifier.take", this.toValueComponent(modif.operation(), value, flag), Component.translatable(attr.getDescriptionId())).withStyle(attr.getStyle(false));
+        }
+
+        return comp.append(this.getDebugInfo(modif, flag));
+    }
+
+    /**
+     * Computes the additional debug information for a given attribute modifier, if the flag {@linkplain TooltipFlag#isAdvanced() is advanced}.
+     * 
+     * @param modif The attribute modifier being converted to a component.
+     * @param flag  The tooltip flag.
+     * @return The debug component, or {@link CommonComponents#EMPTY} if disabled.
+     * @apiNote This information is automatically appended to {@link #toComponent(AttributeModifier, TooltipFlag)}.
+     */
+    default Component getDebugInfo(AttributeModifier modif, TooltipFlag flag) {
+        Component debugInfo = CommonComponents.EMPTY;
+
+        if (flag.isAdvanced()) {
+            // Advanced Tooltips show the underlying operation and the "true" value. We offset MULTIPLY_TOTAL by 1 due to how the operation is calculated.
+            double advValue = (modif.operation() == Operation.ADD_MULTIPLIED_TOTAL ? 1 : 0) + modif.amount();
+            String valueStr = FORMAT.format(advValue);
+            String txt = switch (modif.operation()) {
+                case ADD_VALUE -> advValue > 0 ? String.format("[+%s]", valueStr) : String.format("[%s]", valueStr);
+                case ADD_MULTIPLIED_BASE -> advValue > 0 ? String.format("[+%sx]", valueStr) : String.format("[%sx]", valueStr);
+                case ADD_MULTIPLIED_TOTAL -> String.format("[x%s]", valueStr);
+            };
+            debugInfo = Component.literal(" ").append(Component.literal(txt).withStyle(ChatFormatting.GRAY));
```

Comment:
I think it might be better to just include the space in the above formats rather than having to append components here.

Reply:
I think the existing creates a better information flow. We have a segment that purely creates the debug text, and then one which creates the post-appended (i.e. the with space) component that can be added to the original modifier component.

In the other form, we have to add a leading space to 5 string literals, which make it less obvious why they need the space (and it may be easier to forget it in overrides).